brave/brave-ios
# Fix #6883: Update Swap transactions to use async/await

## Summary of Changes
- Refactor swap transaction store to use async/await for fetching price quote, creating swap transaction. 
- Logic should not be changed, just moved apis to async/await wrappers
    - `fetchPriceQuote` split into `fetchEthPriceQuote` for upcoming Solana Swap
- Resolved/fix all `SwapTokenStoreTests`. Previous tests were not verifying transaction was created, and not creating EIP 1559 transactions.
- Added fetch price quote tests when user edits sell amount and buy amount (from / to values)

This pull request fixes #6883

## Submitter Checklist:


## Test Plan:
1. Create some swap transactions and verify pieces working as expected
    - Logic should not change, just moved apis to async/await wrappers
2. Verify ERC 20 approve transaction can still be created as expected
3. Verify errors display correctly, ex. insufficient liquidity for a token


## Screenshots:



## Reviewer Checklist:

  - `QA/(Yes|No)`
  - `bug` / `enhancement`
